chessington/engine/pieces.py

```diff
@@ -30,17 +30,69 @@ def move_to(self, board, new_square):
 
 
 class Pawn(Piece):
+    def get_available_moves_single_color(self, board: Board, square_1_in_front, square_2_in_front) -> List[Square]:
+        possible_moves = []
+        diagonal_capture_squares = []
+        current_square = board.find_piece(self)
+
+        piece_1_in_front = board.get_piece(square_1_in_front)
+        if piece_1_in_front is None:
+            possible_moves.append(square_1_in_front)
+
+        # pawn did not move yet, it can jump 2 squares
+        if current_square.row == 6 and self.player == Player.BLACK or current_square.row == 1 and self.player == Player.WHITE:
+            piece_2_in_front = board.get_piece(square_2_in_front)
+            if piece_1_in_front is None and piece_2_in_front is None:
+                possible_moves.append(square_2_in_front)
+
+        diagonal_capture_squares.append(Square.at(square_1_in_front.row, current_square.col - 1))
+        diagonal_capture_squares.append(Square.at(square_1_in_front.row, current_square.col + 1))
+
+        for square in diagonal_capture_squares:
+            if square.row < 0 or square.row > 7 or square.col < 0 or square.col > 7:
+                continue
+
+            diagonal_piece = board.get_piece(square)
+            if diagonal_piece is None:
+                continue
+
+            if self.player == diagonal_piece.player:
+                continue
+
+            if isinstance(diagonal_piece, King):
+                continue
+
+            possible_moves.append(square)
+
+        return possible_moves
     """
     A class representing a chess pawn.
     """
     def get_available_moves(self, board) -> List[Square]:
         current_square = board.find_piece(self)
+        possible_moves = []
+
+        # black pawn reached the bottom of the board, no more legal moves
+        if self.player == Player.BLACK and current_square.row == 0:
+            return []
+
+        # white pawn reached the bottom of the board, no more legal moves
+        if self.player == Player.WHITE and current_square.row == 7:
+            return []
```

Comment:
Can you write a function to check if any piece is in bounds?

Reply:
I don't see how would a is_in_bounds function would help me here. I think it would be more useful at the previous comment, - https://github.com/AndreiVoicu04/chessington-python/pull/1/files/7bd27d401600f998e9472c89a643f400ad2a3046#r1665710410